### PR TITLE
Material: Honor missing properties in toJSON() and MaterialLoader.

### PR DESCRIPTION
**Pull request by frading**
this  adds colorWrite to MaterialLoader, and shadowSide to both MaterialLoader and material.toJSON()

**Description**

- material.shadowSide was missing from material.toJSON() and MaterialLoader
- material.colorWrite was missing from material.toJSON()

### CODE REVIEW

**Review by Mugen87 — APPROVED** [review]

File: src/materials/Material.js [hunk 1]
```diff
@@ -255,6 +255,7 @@ Material.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 		}
 
 		if ( this.size !== undefined ) data.size = this.size;
+		if ( this.shadowSide !== undefined ) data.shadowSide = this.shadowSide;
```
Comment (Mugen87) [inline src/materials/Material.js:258]: Wait. I think it should be:
```js
if ( this.shadowSide !== null ) data.shadowSide = this.shadowSide;
```

Comment (Mugen87) [reply src/materials/Material.js:258]: All  materials have a `shadowSide` property so it is never `undefinded`.

Comment (frading) [reply src/materials/Material.js]: Thanks a lot for checking so quickly. And you are very correct! I've updated the branch on my fork, does the PR automatically update from the remote branch or is there something I should do to update this PR?

Comment (Mugen87) [reply src/materials/Material.js]: Yes, updating your branch is sufficient.

Comment (frading) [reply src/materials/Material.js]: Great, thank you

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**